src/components/Router/HistoryRouter.tsx

```diff
@@ -1,173 +1,223 @@
-import { h, Component } from 'preact'
-import {
-  createMemoryHistory,
-  createBrowserHistory,
-  History,
-  LocationListener,
-  MemoryHistory,
-} from 'history'
-
-import { buildStepFinder } from '~utils/steps'
+import { h } from 'preact'
+
+import { buildStepFinder, findFirstIndex } from '~utils/steps'
 import { buildComponentsList } from './StepComponentMap'
 import StepsRouter from './StepsRouter'
 
 import { trackException } from '../../Tracker'
 
-import type { ParsedError, ErrorCallback } from '~types/api'
-import type { ExtendedStepTypes, FlowVariants } from '~types/commons'
+import type { ErrorCallback, ParsedError } from '~types/api'
+import type { FlowVariants, FormattedError } from '~types/commons'
 import type { CaptureKeys } from '~types/redux'
 import type {
-  ComponentStep,
   ChangeFlowProp,
+  ComponentStep,
+  HistoryLocationState,
   HistoryRouterProps,
 } from '~types/routers'
 import type { SdkResponse } from '~types/sdk'
-import type { DocumentTypes } from '~types/steps'
+import type { DocumentTypes, StepConfig } from '~types/steps'
+import { useCallback, useEffect, useState } from 'preact/hooks'
+import { useHistory } from './useHistory'
 
-type FormattedError = {
-  type: 'expired_token' | 'exception'
-  message: string
-}
+type HistoryRouterState = {
+  initialStep: number
+} & HistoryLocationState
+
+const formattedError = ({ response, status }: ParsedError): FormattedError => {
+  const errorResponse = response.error || response || {}
 
-type HistoryLocationState = {
-  step: number
-  flow: FlowVariants
+  const isExpiredTokenError =
+    status === 401 && errorResponse.type === 'expired_token'
+  const type = isExpiredTokenError ? 'expired_token' : 'exception'
+  // `/validate_document` returns a string only. Example: "Token has expired."
+  // Ticket in backlog to update all APIs to use signature similar to main Onfido API
```

Comment:
Please link the ticket

Reply:
This comment is from last year for Standard web SDK, left by Ethan